Sources/ApolloCodegenLib/Frontend/CompilationResult.swift

```diff
@@ -412,4 +462,17 @@ public class CompilationResult: JavaScriptObject {
     }
   }
 
+  public enum IsDeferred: ExpressibleByBooleanLiteral {
```

Comment:
This type is basically the same as `IR.IsDeferred`. I like having them defined in both places even if they're similar. I'm unsure how I feel about sharing a type between the JS frontend (`CompilationResult`) and `IR`. I'm open to suggestions.

Reply:
Yeah, let's leave these separate for now. I don't know if this one needs to be `ExpressibleByBooleanLiteral`, or have the `static func if` though. We're never initializing them directly via literals, are we?

Reply:
Both the `ExpressibleByBooleanLiteral` and the `.if` shortcuts are used in the evaluation function as conveniences. I don't think it's all that bad to include them, they sometimes make for easier reading.